csrc/optimization/alias_analysis.cpp

```diff
@@ -452,4 +453,10 @@ std::string Layout::toString(const int indent_size) const {
   return ss.str();
 }
 
+bool Layout::isCompatibleWith(const Layout& other) const {
+  // Most basic for now.
```

Comment:
:laughing: 

nitpick: what's more advanced than the most basic? I know we can relax contiguity and that's about it?

The function name doesn't suggest the direction of which one between this/other can be relaxed. We might as well just rename it as `isSameAs` to stick better to what it's doing for now.

Reply:
> What's more advanced than the most basic?

I was thinking about broadcast dimensions can be placed differently. 

> I know we can relax contiguity and that's about it?

Thanks -- I totally forgot about that. Yes, it makes sense to have this method be uni-directional. Let me do that in this PR.

Reply:
Done.